### PR TITLE
MVP-100 Breadcrumbs - Router Refactor

### PR DESCRIPTION
**Pull request by SDToews**
## 🎯 Summary

  
[MVP-100](https://citz-imb.atlassian.net/browse/MVP-100)

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](/CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - I have consulted with the team if introducing a new dependency.
> - My changes generate no new warnings.

Refactor router to take advantage of React Router 6 changes

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { RouterProvider } from 'react-router-dom';
+import { router } from './Router';
+
+const AppRouter = () => {
+  return <RouterProvider router={router} />;
```
Comment (BradyMitch) [inline frontend/src/App.tsx:5]: Id like to see this defined in main.tsx

File: frontend/src/constants/packages/package.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { IconProps } from 'components';
+
+export type PackageType = {
```
Comment (BradyMitch) [inline frontend/src/constants/packages/package.ts:3]: Rename file to `types.ts` for consistency

File: frontend/src/pages/Landing.tsx [hunk 3]
```diff
@@ -1,26 +1,24 @@
...
-        />
-        <PackageCard
-          icon="Authenticate"
-          title="SSO Express"
-          pageRoute="/sso-express"
-          summary="An npm package that offers an integration solution for Express applications requiring
-                authentication through the B.C. government's Single Sign-On SSO service."
-        />
+        {packages.map((pkg) => (
+          <Link key={pkg.title} to={pkg.pageRoute} style={{ textDecoration: 'none' }}>
```
Comment (BradyMitch) [inline frontend/src/pages/Landing.tsx:13]: The `PackageCard` component already has an onClick action. We should remove either the `pageRoute` prop for the `PackageCard` or the `Link` component.

**Review by BradyMitch — COMMENTED** [review]
I dont think there should be a `src/types`. The constants and styles you defined here are very specific to a component and dont really resemble anything reusable. It also makes whatever component that uses these less self-contained and less reusable.

File: frontend/src/App.tsx (hunk 1 above)
Comment (SDToews) [reply frontend/src/App.tsx:5]: updated

File: frontend/src/constants/packages/package.ts (hunk 2 above)
Comment (SDToews) [reply frontend/src/constants/packages/package.ts:3]: done

File: frontend/src/pages/Landing.tsx (hunk 3 above)
Comment (SDToews) [reply frontend/src/pages/Landing.tsx:13]: kept the Link component as it allows react router to deal with page changing within the dom

File: frontend/src/components/cards/PackageCard/types.ts [hunk 4]
```diff
@@ -4,5 +4,4 @@ export type PackageCardProps = {
   icon: IconProps['icon'];
   title: string;
   summary: string;
-  pageRoute: string;
```
Comment (BradyMitch) [inline frontend/src/components/cards/PackageCard/types.ts:7]: I think you'll also need to update the component so it doesnt try to use this prop

Comment (SDToews) [reply frontend/src/components/cards/PackageCard/types.ts:7]: I did :)

**Review by BradyMitch — APPROVED** [review]

**Pull request merged**